### PR TITLE
Fixed TrailingCommaRule and StringTemplateRuleFixTest

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- add checking children to support case with comma before comment
- reused ASTFactory

It's part of #1737

### CODE REVIEW

**Review by akuleshov7 — APPROVED** [review]

File: diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter3/TrailingCommaRule.kt [hunk 1]
```diff
@@ -101,9 +101,13 @@ class TrailingCommaRule(configRules: List<RulesConfig>) : DiktatRule(
     }
 
     private fun ASTNode.checkTrailingComma(config: Boolean) {
-        val shouldFix = this.siblings(true).toList().run {
-            !this.map { it.elementType }.contains(COMMA) && this.any { it.isWhiteSpaceWithNewline() || it.isPartOfComment() }
-        }
+        val noCommaInSiblings = siblings(true).toList()
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter3/TrailingCommaRule.kt:104]: the only this I suggest is work with `toSeq` here

Comment (nulls) [reply diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter3/TrailingCommaRule.kt:104]: do you mean `toSet`?

Comment (nulls) [reply diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter3/TrailingCommaRule.kt]: changed to `toSet` + fixed `StringTemplateRuleFixTest`

Comment (nulls) [reply diktat-rules/src/main/kotlin/com/saveourtool/diktat/ruleset/rules/chapter3/TrailingCommaRule.kt]: discussed offline -- it's not possible or will be unreadable

**Pull request merged**